app/backend/src/clj/teet/integration/vektorio/vektorio_client.clj

```diff
@@ -89,12 +99,18 @@
 
 (defn add-user-to-project!
   "Add the given user to the vektorio project"
-  [vektor-conf project-id user-id]
+  [conn user vektor-conf project-id user-id]
   (assert (some? project-id) "Must specify the project to which the user is added")
   (assert (some? user-id) "Must specify the user which is added to the project")
   (log/info "Adding vektorio user" user-id "to vektorio project:" project-id)
-  (vektor-post! vektor-conf {:endpoint (str "projects/" project-id "/users")
-                             :payload {:userId user-id}}))
+  (when
+    (empty?
+      (d/q '[:find ?pid :where [?u :vektorio/granted-projects ?pid] :in $ ?u ?pid]
+         (d/db conn) (user-model/user-ref user) project-id))
```

Comment:
This addition introduces a dependency to datomic. Let's keep this namespace purely a wrapper for the Vektorio HTTP API and instead pass make this check in the namespace where `add-user-to-project!` is called.